src/mscorlib/shared/System/Reflection/ReflectionTypeLoadException.cs

```diff
@@ -62,11 +65,16 @@ public override string Message
         public override string ToString()
         {
             StringBuilder text = new StringBuilder();
-            text.AppendLine(SR.ReflectionTypeLoad_LoadFailed);
-
-            foreach (Exception e in LoaderExceptions)
+            text.AppendLine(base.ToString());
+            if (LoaderExceptions != null)
```

Comment:
Nit: It'd be nice if the check was moved to the top of the method (include the length check) and early exit to match the same flow as in `Message`.

A few other things:
 - It might be nice to avoid the extra NewLine that will be at the end of the resulting string.
 - Just a nit, but `LoaderExceptions` could be cached in a local.
 - The initial string can be passed to the `StringBuilder`'s constructor.

Altogether, something like:

```c#
        public override string Message
        {
            get
            {
                Exceptions[] exceptions = LoaderExceptions;
                if (exceptions == null || exceptions.Length == 0)
                {
                    return base.Message;
                }

                var text = new StringBuilder(base.Message);
                foreach (Exception e in exceptions)
                {
                    if (e != null)
                    {
                        text.AppendLine();
                        text.Append(e.Message);
                    }
                }
                return text.ToString();
            }
        }

        public override string ToString()
        {
            Exceptions[] exceptions = LoaderExceptions;
            if (exceptions == null || exceptions.Length == 0)
            {
                return base.ToString();
            }

            var text = new StringBuilder(base.ToString());
            foreach (Exception e in exceptions)
            {
                if (e != null)
                {
                    text.AppendLine();
                    text.Append(e.ToString());
                }
            }
            return text.ToString();
        }
```

Reply:
This is not a hot path, I don't see the value of caching the field access.

Reply:
> I don't see the value of caching the field access.

It tends to make the code smaller.

Reply:
Also, this is not a plain field access. It is a property access that is harder for JIT to optimize out.

E.g. if I change LoaderExceptions to be cached in a local in what the currently checked in implementation of Message property, the x64 native code will get `~6%` smaller (from 220 bytes to 206 bytes).